update-common/src/artifacts/update_plan.rs

```diff
@@ -357,13 +374,25 @@ impl<'a> UpdatePlanBuilder<'a> {
             data.extend_from_slice(&chunk);
         }
 
-        let (artifact_id, bootloader_sign) =
-            read_hubris_sign_from_archive(artifact_id, data.clone())?;
+        let (artifact_id, bootloader_caboose) =
+            read_hubris_caboose_from_archive(artifact_id, data.clone(), true)?;
 
-        self.rot_by_sign
-            .entry(RotSignData { kind: artifact_kind, sign: bootloader_sign })
-            .or_default()
-            .push(artifact_id.clone());
+        // We restrict the bootloader to exactly one entry per (kind, signature)
+        match self.rot_by_sign.entry(RotSignData {
+            kind: artifact_kind,
+            sign: bootloader_caboose.sign.expect("required SIGN in caboose"),
```

Comment:
Where is this requirement enforced? Wondering if a lack of `sign` would have already failed somewhere above, or if it's possible for a malformed image to get here and cause a panic.

(Same comment on the other `.sign.expect("..")`s below)

Reply:
Oh I see, `read_hubris_caboose_from_archive(..., true)` requires sign to be present or fails. That seems easy to miss but I don't have a great alternative to suggest. I guess we could have two different `read_hubris_caboose_from_archive_{with,without}_sign` that return different types (one with a non-optional `sign` field, the other without)? I think I'd slightly prefer that but if it's a bunch of busywork and/or you don't think it's worth it, I won't object.

Reply:
That's a good suggestion as I really didn't like the `bool` parameter